torch/linalg/__init__.py

```diff
@@ -813,6 +813,102 @@
     https://www.netlib.org/lapack/lug/node128.html
 """)
 
+ldl_factor_ex = _add_docstr(_linalg.linalg_ldl_factor_ex, r"""
+linalg.ldl_factor_ex(A, *, upper=False, hermitian=False, check_errors=False, out=None) -> (Tensor, Tensor, Tensor)
+
+Computes the LDL decomposition of a Hermitian/symmetric indefinite matrix.
+When :attr:`A` is complex valued it can be Hermitian (:attr:`hermitian` = `True`)
+or symmetric (:attr:`hermitian` = `False`).
```

Comment:
```suggestion
or symmetric (:attr:`hermitian` = `False`). If :attr:`A` is real, it is assumed to be a symmetric matrix.
```

Reply:
What does "assumed" mean here? Do you want to have mentions in the docs that only lower/upper triangular part of the inputs is used in the computation?

Reply:
What I meant to convey here is that in the real case the hermitian attribute does not change anything. You can add to the proposed sentence "irrespective of the value of ``:attr:`hermitian` ``" if you think that makes it clearer.

So, if we go with the user of the `lower` argument (which I'd prefer not to) we would need to describe exactly how this function works. In particular we'd need to describe those nuances, and probably say that the gradient is upper/lower triangular. If the inputs are assumed to be symmetric / Hermitian (no `lower` argument), then we can skip all this.